build-tools/package.json

```diff
@@ -84,7 +84,7 @@
 		"rimraf": "^4.4.1",
 		"run-script-os": "^1.1.6",
 		"syncpack": "^9.8.6",
-		"typescript": "~4.5.5"
+		"typescript": "~5.1.6"
```

Comment:
As we've observed, the file hashes stored in '*.tsbuildinfo' can vary by compiler version, causing incremental builds to conservatively rebuild everything.

In a worst case scenario, this changes with each release of the compiler, forcing us to update 'fluid-builds' each time we move forward (similar to how we must bump api-extractor.)

Reply:
Do you know if fluid-build uses the version of TypeScript that's installed local to the package? If not, it probably should. If it does, though, maybe we should lock the TypeScript version we use to an exact version so we never see unexpected breaks.

Reply:
> we should lock the TypeScript version we use to an exact version

This would be a separate change to the client release group, if we want to do it. Not needed in this PR.

Reply:
I believe it's a bit mixed: when fluid-build calls into the TypeScript compiler, it's running code from its local node_modules folder.  When 'fluid-build' invokes the TypeScript compiler to do the actual compilation, it spawns a process to execute a shell command, which will use the compiler in the target package's node_modules folder.

Since 'fluid-build' is already semi-coupled to a TypeScript version, something that might be interesting to try in a future FHL would be for fluid-build to invoke its local TypeScript compiler via code rather than spawning a shell process with a separate node instance.  (Years ago, this technique was used by WebPack to significantly speed up hot compilation by reusing the JITted compiler already in memory rather than booting a cold node process.)

If we did this, we could also probably stop inspecting tsbuildinfo in 'fluid-build', and just let TypeScript figure out that no files have changed.

I'm actually about to bump the TypeScript version in client, so it would be a good time to lock to a specific version, although I suspect our current "~" range is _probably_ stable enough in practice.